src/patternfly/components/OverflowMenu/overflow-menu.scss

```diff
@@ -1,89 +1,51 @@
-// @debug $overflow-menu; // check your variable names located in src/patternfly/sass-utilities/component-namespaces.scss
+:where(:root),
+:where(.#{$overflow-menu}) {
+  --#{$overflow-menu}--ColumnGap: var(--pf-t--global--spacer--sm);
 
-.#{$overflow-menu} {
-  // Base spacer - shared value
-  --#{$overflow-menu}--spacer--base: var(--#{$pf-global}--spacer--md);
-
-  // Spacer values
-  --#{$overflow-menu}--spacer: var(--#{$pf-global}--spacer--sm);
-  --#{$overflow-menu}__group--spacer: var(--#{$overflow-menu}--spacer--base);
-  --#{$overflow-menu}__item--spacer: var(--#{$overflow-menu}--spacer--base);
-  --#{$overflow-menu}--c-divider--m-vertical--spacer: var(--#{$overflow-menu}--spacer--base);
-
-  // Button group
-  --#{$overflow-menu}__group--m-button-group--spacer: var(--#{$overflow-menu}--spacer--base);
-  --#{$overflow-menu}__group--m-button-group--space-items: var(--#{$pf-global}--spacer--sm);
+  // * Overflow menu icon button group
+  --#{$overflow-menu}__group--m-icon-button-group--ColumnGap: 0;
+}
 
-  // Icon button group
-  --#{$overflow-menu}__group--m-icon-button-group--spacer: var(--#{$overflow-menu}--spacer--base);
-  --#{$overflow-menu}__group--m-icon-button-group--space-items: 0;
+// - Overflow menu - Overflow menu content - Overflow menu group
+.#{$overflow-menu},
+.#{$overflow-menu}__content,
+.#{$overflow-menu}__group {
+  display: flex;
+}
 
-  display: inline-flex;
-  align-items: center;
+// - Overflow menu
+.#{$overflow-menu} {
+  column-gap: var(--#{$overflow-menu}--ColumnGap);
+  width: fit-content;
 }
 
-// Content
+// - Overflow menu content
 .#{$overflow-menu}__content {
-  display: flex;
-  align-items: center;
+  align-items: start;
 }
 
-// Group
+// - Overflow menu group
 .#{$overflow-menu}__group {
-  --#{$overflow-menu}--spacer: var(--#{$overflow-menu}__group--spacer);
-
-  display: flex;
-  align-items: center;
+  column-gap: var(--#{$overflow-menu}__group--ColumnGap, var(--#{$overflow-menu}--ColumnGap));
 
-  // Button group
+  // - Overflow menu button group
   &.pf-m-button-group {
-    --#{$overflow-menu}--spacer: var(--#{$overflow-menu}__group--m-button-group--spacer);
-
-    > * {
-      --#{$overflow-menu}--spacer: var(--#{$overflow-menu}__group--m-button-group--space-items);
-    }
+    column-gap: var(--#{$overflow-menu}__group--m-button-group--ColumnGap, var(--#{$overflow-menu}--ColumnGap));
   }
 
-  // Icon button group
+  // - Overflow menu icon button group
   &.pf-m-icon-button-group {
-    --#{$overflow-menu}--spacer: var(--#{$overflow-menu}__group--m-icon-button-group--spacer);
-
-    > * {
-      --#{$overflow-menu}--spacer: var(--#{$overflow-menu}__group--m-icon-button-group--space-items);
-    }
+    column-gap: var(--#{$overflow-menu}__group--m-icon-button-group--ColumnGap, var(--#{$overflow-menu}--ColumnGap));
```

Comment:
Do we need the fallback since the group--m-icon--button-group--columngap is defined?

Reply:
Nice catch! We don't.